scripts/apps/authoring/authoring/previewModal.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {Modal} from 'core/ui/components/Modal/Modal';
+import {ModalHeader} from 'core/ui/components/Modal/ModalHeader';
+import {ModalBody} from 'core/ui/components/Modal/ModalBody';
+import {ModalFooter} from 'core/ui/components/Modal/ModalFooter';
+
+function getFormattedDocument(url) {
+    return new Promise((resolve) => {
+        var xhr = new XMLHttpRequest();
+
+        xhr.addEventListener('load', function() {
+            resolve({
+                fomattedDocument: this.responseText,
+                documentContentType: this.getResponseHeader('content-type'),
+            });
+        });
+
+        xhr.open('GET', url);
+        xhr.send();
+    });
+}
+
+export class PreviewModal extends React.Component {
+    openPreviewForItem(subscriberId, format, endpointUrl) {
+        const {urls, documentId, gettext} = this.props;
+
+        const nextWindow = window.open();
+
+        nextWindow.document.body.innerText = gettext('Loading preview');
+
+        const url = urls.item('format-document-for-preview')
+            + `?subscriber_id=${subscriberId}&formatter=${format}&document_id=${documentId}`;
```

Comment:
I think it should send full item, otherwise you would have to save before you can preview

Reply:
I don't have a full item nearby. There is `scope.item`, but it has missing properties compared to the object in the database. Let's require saving for now. I've put a message asking to save it in case previewing of an unsaved item is attempted.